src/lib/plugin_apis/fs.api

```diff
@@ -1188,4 +1269,84 @@ gboolean bd_fs_ntfs_resize (const gchar *device, guint64 new_size, GError **erro
  */
 BDFSNtfsInfo* bd_fs_ntfs_get_info (const gchar *device, GError **error);
 
+/**
+ * bd_fs_f2fs_mkfs:
+ * @device: the device to create a new f2fs fs on
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the creation (right now
+ *                                                 passed to the 'mkfs.f2fs' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether a new f2fs fs was successfully created on @device or not
+ *
+ * Tech category: %BD_FS_TECH_F2FS-%BD_FS_TECH_MODE_MKFS
+ */
+gboolean bd_fs_f2fs_mkfs (const gchar *device, const BDExtraArg **extra, GError **error);
+
+/**
+ * bd_fs_f2fs_wipe:
+ * @device: the device to wipe an f2fs signature from
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether an f2fs signature was successfully wiped from the @device or
+ *          not
+ *
+ * Tech category: %BD_FS_TECH_F2FS-%BD_FS_TECH_MODE_WIPE
+ */
+gboolean bd_fs_f2fs_wipe (const gchar *device, GError **error);
+
+/**
+ * bd_fs_f2fs_check:
+ * @device: the device containing the file system to check
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the repair (right now
+ *                                                 passed to the 'fsck.f2fs' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether an f2fs file system on the @device is clean or not
+ *
+ * Tech category: %BD_FS_TECH_F2FS-%BD_FS_TECH_MODE_CHECK
+ */
+gboolean bd_fs_f2fs_check (const gchar *device, const BDExtraArg **extra, GError **error);
```

Comment:
Why do we want to support `**extra` for `f2fs_check()` and `f2fs_repair()`?

Reply:
We already have extra arguments for these for other filesystems like vfat and ext and it is possible there will be more arguments for `fsck.f2fs` in the future (they are still adding new features).